### DISCUSSION TITLE
COMO COMEÇO NA ÁREA?

### DISCUSSION THREAD

**Discussion by juaum1st** [frontendbr/forum#2506] [Discussão]
Opa galera, meu nome é João, atualmente tenho 17 anos e to em busca de conhecimento na área da programação, só que não tenho um caminho certo para seguir, não sei praticamente nada, mas tudo tem o seu começo, se alguém puder me ajudar a como começar, vai me ajudar muito, eu tenho um curso na Udemy que comprei essa semana de Javascript, mas não entendi muito bem, acho que tenho que ver algoritmos e lógica. 

Por onde eu começo?

**Comment by nao-e-oEdu** [comment]
Salve João! Eu sou o Eduardo, eu tenho 16 anos e eu comecei a programar aos 14, com JS tmb, até hj eu não sei como se começa alguma coisa, eu só vou fazendo da forma que sei e pego o trem andando, às vezes me aparece um problema, ou eu me faço um desafio que eu não sei concluir, aí vou pro youtube, pro chatGPT, dou uma pesquisada em qualquer lugar mesmo, mas quando eu fui começar na programação eu comecei baixando o GrassHoper, é um app pra celular, que vc conecta sua conta do Google e ele te ensina JS desde o básico até o início do intermediário. Eu não sei como está hj, pq quando eu parei de usar ainda estava em desenvolvimento, mas foi um ótimo primeiro passo pra mim. Depois de um tempo, quando eu fui pro ensino médio eu entrei na etec pra fazer um curso de programação, e desde então são nas lições dos professores que eu me baseio pra começar algo novo.

**Reply by juaum1st** [reply]
muito obrigado pelo comentário, vou ver esse grasshoper

**Comment by mateuscardoso99** [comment]
Opa! eu posso te responder com base no que aconteceu comigo, eu comecei com 19 anos a programar e estudei algoritmo e programação em C pouca coisa, sem praticar muito e fui direto estudar frameworks como react, spring, vue, laravel sabendo pouco de java, php, js etc.. esse foi o meu maior erro, pois sempre ficava copiando soluções mas tinha pouca capacidade de criar soluções. É preciso saber os fundamentos, como um computador funciona, como a memória funciona, lógica de programação, estruturas de dados como vetor, matriz, pilha, fila, árvore, grafos, ordenação, ponteiros... quando entende como as coisas funcionam e tem bom conhecimento de estruturas de dados e tem lógica na cabeça, quando digo ter lógica na cabeça quero dizer que tu vê um problema e como tu já sabe manipular estruturas de dados, laços de repetição, conhece os fundamentos etc.. tu já consegue criar soluções na cabeça e consegue criar as tuas próprias soluções, tu se sente mais confortável em programar, já conhece os caminhos..

**Reply by juaum1st** [reply]
muito obrigado pelo comentário mateus, estava começando pelo js, mas fiquei doido, pois pulei a parte de lógica, vou fazer isso para não ficar perdido, obrigado!!!

**Comment by CassiusHorvath** [comment]
Podes aprender sozinho, um pouco aqui um pouco lá mas isso regra geral não é muito produtivo. Na minha opinião podes fazer algum curso da Udemy, são acessíveis e muito completos. Alguns até dão aula de algoritmo o que acho fundamental. Com issso adquires a base inicial para começar a trilhar a sua carreira. Se tua intenção é sair do Brasil sugiro fazer uma universidade na área, ao contrário do Brasil aqui fora 90% ou mais das vagas boas exigem nível superior. Depois de dominar bem JS, HTML e CSS, estuda React ou Angular.. ou mesmo os dois, pois React é uma library enquanto Angular é um framework. Nos EUA usam muito react enquanto que na Europa, principalmente na Alemanha o Angular tem muito espaço. Mais lá na frente é interessante começar a pensar em aprender um pouco de backend pois o mercado por vezes exige que o dev seja pelo menos um pouco full-stack. Conhecimentos sobre Scrum, GIT, Arquiteturas, Design Patterns, S.O.L.I.D., rest, são coisas que no futuro deves começar a pesquisar e estudar. A vida de um developer é aperfeiçoamento contínuo, você NUNCA vai parar de estudar e aprender coisas novas, se parar de estudar vai fracassar, vai ficar no meio do caminho, é a verdade nua e crua. Se está de bem com isso, seja bem vindo ao mágico mundo da engenharia de software!!!!
Basicamente resumi o eu fiz na minha vida como dev, não necessariamente serve para outros mas essa fórmula deu certo para mim.
Um último aviso, se ainda não fala, aprenda inglês decentemente o mais rápido possível. Para uma carreira internacional NO MÍNIMO tem que falar inglês nível B2, preferencialmente C1 ou acima. Na Europa ainda será muito bem vindo se for fluente em francês e /ou alemão. 
Espero ter ajudado e qualquer coisa só perguntar diretamente.

**Reply by juaum1st** [reply]
MUITO obrigado, tenho uma base agora, já que isso tudo tava virando uma bagunça para mim, teno um curso de javascript na Udemy, estou começando agora, muito obrigado pelo apoio meu amigo, agradeço de coração.

**Comment by Emanoellima-dev** [comment]
Estude lógica de programação. Faça seus primeiros projetos. e mesmo que sejam projetos bem simples, o importante é você praticar bem.

**Comment by nazarepiedady** [comment]
Todos os que forem começar a programar como carreira ou como forma de entretenimento, deverão passar pelos conhecimentos da Ciência da Computação e Fundamentos da Programação de Computadores, se não perder-se-ão no oceano de conteúdos inúteis e repetitivos.
